powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/v3/PowerAuthServiceImpl.java

```diff
@@ -1409,23 +1409,23 @@ public ListActivationFlagsResponse listActivationFlags(ListActivationFlagsReques
 
     @Override
     @Transactional
```

Comment:
Rollback strategy?

Reply:
The current rollback handling should be fine:
- exceptions are thrown before any persistence to database
- in case of unexpected runtime errors the rollback is applied (`RuntimeException` and `Error` exceptions are always propagated in `PowerAuthServiceImpl`)
- we use special handling of rollbacks only after some data has been written to database (e.g. an unexpected error after `initActivation`), no need to use it here

At any case I found that several `@Transactional` annotations were missing, so I fixed this.